### PR TITLE
Travel between zones!

### PR DESCRIPTION
**Pull request by b1gd3vd0g**
-Couple updates to AssetManager because it is now getting reset (to BAREBONES_DL_Q) upon the loading of every zone.
-The village.main and village.field zones are not pretty, but they exist. You may walk between them.
-A Border is an invisible entity which exists in the midground, OUTSIDE the boundaries of a zone. When chad collides with one, he is transported to another Zone.
-Game keeps track of ZONE and LAST_ZONE, in order to determine spawn point.

### CODE REVIEW

File: js/classes/asset_manager.js
Comment (NathanHinthorne) [inline js/classes/asset_manager.js:1]: Love the barebones changes. Now we're loading in just the essentials we need as we transition between zones.

File: js/classes/zone.js
Comment (NathanHinthorne) [inline js/classes/zone.js:1]: This new layout for each dimension looks really well structured.

**Review by NathanHinthorne — APPROVED** [review]

File: js/classes/asset_manager.js [hunk 1]
```diff
@@ -172,5 +174,25 @@ class AssetManager {
             }
         }
     };
+
+    /**
+     * The BAREBONES_DL_Q is an array of strings, filepaths to the most essential assets in our game.
+     * Those so essential that it would be tedious and more error prone to have to load them in every zone.
+     * this.downloadQueue is reset to this every time that the ASSET_MGR is refreshed.
+     */
+    static get BAREBONES_DL_Q() {
```
Comment (calebkrauter) [inline js/classes/asset_manager.js:183]: I really like this. This type of array access is new to me but it is awesome!

File: js/classes/entities/background/decoration.js [hunk 2]
```diff
@@ -9,56 +9,97 @@ class Decoration {
...
-        const properties = Decoration.PROPERTY_TABLE[type];
-        const size = properties.SIZE;
-
-        this.scale = properties.SCALE;
-        this.pos = new Vector(bottomLeftPos.x, bottomLeftPos.y - size.y * this.scale);
-        this.animator = new Animator(properties.SPRITESHEET, properties.SPRITESHEET_START_POS, 
-            size, properties.FRAME_COUNT, properties.FRAME_DURATION);
+        this.type = type;
+        this.pos = new Vector(bottomLeftPos.x, bottomLeftPos.y - this.type.SIZE.y * this.type.SCALE);
+        this.animator = new Animator(this.type.SPRITESHEET, this.type.SPRITESHEET_START_POS,
```
Comment (calebkrauter) [inline js/classes/entities/background/decoration.js:19]: Nice work making this readable!

File: js/classes/entities/background/decoration.js [hunk 3]
```diff
@@ -9,56 +9,97 @@ class Decoration {
...
+                }
             },
-            Tree: {
-                SPRITESHEET: "./sprites/tree.png",
-                SPRITESHEET_START_POS: new Vector(0, 0),
-                FRAME_COUNT: 1,
-                FRAME_DURATION: 1,
-                SIZE: new Vector(64, 64),
-                SCALE: 8,
+            trees: {
```
Comment (calebkrauter) [inline js/classes/entities/background/decoration.js:94]: Didn't realize you could make this so flexible by having trees and specific ones that stem from it. Is this JSON? I like it!

File: js/classes/entities/midground/border.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/**
+ * A border is an INVISIBLE ENTITY which allows Chad to enter a new zone. 
+ * It should NEVER be placed within the boundaries of a zone, but rather just outside.
+ * It should be impossible to miss.
+ */
+class Border {
```
Comment (calebkrauter) [inline js/classes/entities/midground/border.js:6]: GENIUS @B1GD3VD0G !

File: js/classes/zone.js [hunk 5]
```diff
@@ -115,23 +130,23 @@ class Zone {
 
             },
             playground: {
-                caleb:          new Zone(zeros, defaultMaxBlock, loadPlaygroundCaleb),
-                devin:          new Zone(zeros, defaultMaxBlock, loadPlaygroundDevin),
-                everybody:      new Zone(zeros, defaultMaxBlock, loadPlaygroundEverybody),
-                nathan:         new Zone(zeros, defaultMaxBlock, loadPlaygroundNathan),
-                trae:           new Zone(zeros, defaultMaxBlock, loadPlaygroundTrae)
+                caleb:          new Zone(zeros, defaultMaxBlock, loadPlaygroundCaleb, "Club Caleb"),
```
Comment (calebkrauter) [inline js/classes/zone.js:133]: Haha nice! Love it.

File: js/scripts/zone_loaders/village_zone_loaders.js [hunk 6]
```diff
@@ -38,21 +70,41 @@ const loadVillageInsideCave = () => {
...
-        const blockPos = new Vector(5, 20);
-        CHAD.pos = Vector.blockToWorldSpace(blockPos);
+
+        // Place chad.
+
+        if (LAST_ZONE === null) { // We've just started the game.
+            // Spawn in middle.
+            const blockPos = new Vector(50, 20);
+            CHAD.pos = Vector.blockToWorldSpace(blockPos);
+
```
Comment (calebkrauter) [inline js/scripts/zone_loaders/village_zone_loaders.js:93]: Love this! Very straightforward!

**Review by calebkrauter — APPROVED** [review]
Looks great! Love it!

**Pull request merged**